src/components/CommonComponents/Pagination/Page.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { MouseEvent, ReactElement, FC } from 'react';
+import classnames from 'classnames';
+import { LocalizedLink as Link } from 'gatsby-theme-i18n';
+import { FormattedMessage } from 'react-intl';
+import { PageModel, PageType } from './util/buildPaginationModel';
+
+interface Props {
+  className: string;
+  page: PageModel;
+  hrefBuilder: (n: number) => string;
+  onPageChange?: (e: MouseEvent) => void;
+}
+
+const getPagerContent = (page: PageModel): string | ReactElement => {
+  const content = {
```

Comment:
This constant could reside outside of the function. E.g.:

```jsx

const PAGINATION_TYPES = {
    [PageType.PREV]: <FormattedMessage id="components.pagination.previous" />,
    [PageType.NEXT]: <FormattedMessage id="components.pagination.next" />,
    [PageType.NUM]: String(page.num),
    [PageType.BREAK]: '…',
  };

const getPaginationType = (page: PageModel): JSX.Element => content[page.type]
```

I don't like the idea that a function can both return a string and a JSX.Element, that sounds inconsistent. Even more that if your input is a `PageModel` it should be guaranteed that only a valid entry will be returned.

Reply:
fixed